rbac/management/utils.py

```diff
@@ -279,6 +279,12 @@ def validate_group_name(name):
         raise serializers.ValidationError({key: _(message)})
 
 
+def check_duplicate_entry(grp_name, tenant_id):
+    """Check for duplicate group per tenant."""
+    if Group.objects.filter(tenant=tenant_id).filter(name=grp_name):
+        raise serializers.ValidationError({"Group" + " " + grp_name + " " + "already exists": "Group already exists"})
+
+
```

Comment:
the `source` in the response should be in this case only `group` and message with the duplicate group name as `detail` property, so the response can look like this
```
 {
    "errors": [
        {
            "detail": "Group 'ABC' already exists.",
            "source": "Group ABC already exists",
            "status": "400"
        }
    ]
}
```

you can use [f strings](https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings) to increase the readability, something like `raise serializers.ValidationError({"group": f"Group '{grp_name}' already exists."})`